Added ArrayList task solution

ksenixen opened the pull request.

overpathz commented on src/main/java/core/basesyntax/ArrayList.java:
```diff
@@ -1,48 +1,166 @@
+    private int capacity;
+
+    public ArrayList() {
+        this.elementData = EMPTY_INITIAL_ARRAY;
```
We should create here an array with default capacity

overpathz commented on src/main/java/core/basesyntax/ArrayList.java:
```diff
@@ -1,48 +1,166 @@
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_CAPACITY = 10;
+    public static final double DEFAULT_CAPACITY_K = 1.5;
```
What is K?

overpathz commented on src/main/java/core/basesyntax/ArrayList.java:
```diff
@@ -1,48 +1,166 @@
+        return size == 0;
+    }
+
+    private Object[] grow(double k) {
```
Please, give the more appropriate name for the variable

overpathz commented on src/main/java/core/basesyntax/ArrayList.java:
```diff
@@ -1,48 +1,166 @@
+        }
+        boolean elementExist = false;
+        T oldElement = null;
+        int index = 0;
```
We can move this to the separate method: indexOf(element)
Then, use remove(int index) to remove such element(sure, and perform check for index)

overpathz commented on src/main/java/core/basesyntax/ArrayList.java:
```diff
@@ -1,48 +1,166 @@
+                capacity = 0;
+                elementData = EMPTY_INITIAL_ARRAY;
+            } else if (capacity / size >= DEFAULT_CAPACITY_K) {
+                elementData = resize();
```
instead of resizing and creating a lot of data, we can just shift our elements:
System.arraycopy(elements, startIndex + 1, elements, startIndex, size - startIndex - 1);

overpathz reviewed (changes requested):
It's a very nice that you have looked to native array list implementation.
Need provide some changes to approve

ksenixen commented:
Added indexOf method. Modified add, remove to shift elements instead of resizing. Modified remove by element to use remove(int index) and addAll to use add method
Changed ArrayList constructor
Removed using of K variables

olekskov reviewed (approved):
Good job
